text/0001-PXE-Booting-UEFI.md

```diff
@@ -62,19 +62,24 @@ Foreman Proxy must be enhanced with new PXE variant "Pxegrub2" to allow deployin
 
 ### [Foreman core #12634](http://projects.theforeman.org/issues/12634) - New HW Model flags pxe_template and loader
 
-Hardware model will be enhanced with two new flags: PXE template specifies PXE template kind and variant pair to use. Until today, this was hardcoded to PXELinux or PXEGrub by selected operating system. Optionally, template variant flag from OS could be simply dropped because PXE template implies variant to use and there are no variants which share the same configuration syntax (this is an open item).
+Foreman will support both BIOS and UEFI systems and it is required to associate extra information for each indidivual host (or hostgroup) in order to provide TFTP "filename" option correctly. It can be set per instance via new Host flag (free-form dropdown field) called "PXE Loader" (`pxe_loader` model attribute) with possible values:
 
-Orchestration TFTP code must be modified to use multiple "KIND LOADER default local boot" and "KIND LOADER global default" templates. Local and global boot templates must be added for Grub2 new kind. Naming convention will be:
+* pxelinux.0 (default value)
```

Comment:
> Does that sound good @lzap ? My main goal with this is to not complicate the host form further.

That looks like error prone and fragile workflow to me, I don't like it.
Reasons:

* Users who will have most of their infrastructure on the loader we will
hardcode to load as the second would need to wait from now on.

* How long should we wait? Some servers takes ages to initialize, this
will make the proces twice as slow as it was.

* Some servers might get stuck in this reboot loop easily - EFI firmware
loaders do random things on failures (e.g. halts system completely).

* How about servers without BMC? I assume your suggestion would only
work with BMC.

* This breaks discovered hosts.

-- 
Later,
 Lukas #lzap Zapletal